src/Compilation.zig

```diff
@@ -978,6 +979,9 @@ pub fn create(gpa: Allocator, options: InitOptions) !*Compilation {
             if (!build_options.have_llvm)
                 break :blk false;
 
+            if (!std.process.can_spawn)
+                break :blk false;
```

Comment:
This should be removed. We can still invoke LLD and Clang even if we cannot spawn processes.

Reply:
Hmm, I had assumed that LLD/Clang couldn't be safely called multiple times, or may never return, due to [a comment in the linker](https://github.com/ziglang/zig/blob/71e0cca7a7957e2f024d2985318e478aa6fb1451/src/link/Elf.zig#L1613-L1615)
> LLD does not behave properly as a library - it calls exit() and does not reset all global data between invocations.

Is that incorrect after all?

Reply:
It mostly works but it's buggy. I think we should print a warning when invoking LLD for the second time in the same process execution.

Reply:
Check out 72f6c6e6345030392a3f8cac79862d58359f1e76

Reply:
Great, thanks for the reference. So the preference is to support these calls with warnings, despite the dangers, rather than to block this functionality completely - Makes sense.

I'll go ahead and add non-terminating equivalents of `punt_to_clang` and friends to `src/codegen/llvm.zig`, and update the compiler and linkers to use those on non-spawning OS's

Let me know if you see any issues with that

Reply:
Sounds mostly good. One suggestion:

Extract the part in `punt_to_clang` that doesn't call exit() into a function called `clangMain`. Make sure the arena memory is freed because we're not going to call exit(). Make *that* function public, leaving `punt_to_clang` private. It can live in main.zig, that's a better place than src/codegen/llvm.zig, which is specifically for lowering AIR to LLVM IR.